x-pack/plugins/fleet/server/services/agents/crud.test.ts

```diff
@@ -192,4 +194,85 @@ describe('Agents CRUD test', () => {
       });
     });
   });
+
+  describe('processAgentsInBatches', () => {
+    const mockProcessAgents = (agents: Agent[]) =>
+      Promise.resolve({ items: agents.map((agent) => ({ id: agent.id, success: true })) });
+    it('should return results for multiple batches', async () => {
+      searchMock
+        .mockImplementationOnce(() => Promise.resolve(getEsResponse(['1', '2'], 3)))
+        .mockImplementationOnce(() => Promise.resolve(getEsResponse(['3'], 3)));
+
+      const response = await processAgentsInBatches(
+        esClientMock,
+        {
+          kuery: 'active:true',
+          perPage: 2,
```

Comment:
We pass `perPage: 2` here, yet expect the response to send back 3 `items`? Confused by this.

Reply:
`perPage` means here the `batchSize`, I can rename it for clarity.